### ISSUE TITLE
[Feature Request] Include Test::Minion subclass/role of Test::Mojo

### ISSUE THREAD

**Issue by kiwiroy** [mojolicious/minion#95]
It would be nice to have a `Test::Minion` module to use with apps that `plugin Minion => {};`.

Something to simplify enqueuing, performing and testing of tasks. I started a branch https://github.com/mojolicious/minion/compare/master...kiwiroy:test-minion and was thinking something like

```perl
$t = Test::Minion->new;
$t->task_exists('add')
  ->task_exists('convert')
  ->task_is(add => [1, 1], 2, $desc)
  ->task_is(convert => ['image.jpg'], 'image.png', $desc)
  ->perform_jobs;
```

**Comment by kraih** [comment]
Interesting idea, `perform_jobs` should probably be `perform_jobs_ok`.

**Comment by kraih** [comment]
And i wouldn't mind shipping a role with Minion either.
```
my $t = Test::Mojo->new(...)->with_roles('+Minion');
```

**Comment by kraih** [comment]
But i do think this is currently blocked by `Test::Mojo::_test` being a private API. We'll have to figure out a better alternative for that in Mojolicious first.

**Comment by kiwiroy** [comment]
Yeah, I missed the `perform_jobs_ok` - that's kinda obvious now. There's also missing increments of `$Test::Builder::Level`. Shipping `Test::Minion` means depending `Role::Tiny`.

**Comment by kraih** [comment]
And `task_is` looks more like it should be `enqueue_ok`.

**Comment by rabbiveesh** [comment]
while you're at it, you probably want to be able to assert things against the result of the job. Sometimes you're testing that it fails, sometimes that it succeeded, etc.

**Comment by kraih** [comment]
`Test::Mojo` has been updated to make this easier. https://github.com/mojolicious/mojo/compare/4f1fb79883e5...3723973fd500#diff-ee9e1eca670f561769d858a66424416e

**Comment by kiwiroy** [comment]
Good work on `Test::Mojo`. I've updated the branch. Waiting on release before PR. https://github.com/mojolicious/minion/compare/master...kiwiroy:test-minion#diff-327a0d12729d8cb0b9a10520244a9b22R7

@rabbiveesh `enqueue_ok` requires a result or `subtest` block that will be tested by `perform_jobs_ok`. The subtest can test any state of `$job` not just `$job->info->{result}`. Additionally, `job_is` can do the same for any children (jobs enqueued by others).

**Comment by kraih** [comment]
@kiwiroy Please be aware that your branch will need *a lot* of cleaning up before it can be considered.

**Comment by kraih** [comment]
I'm starting to believe that this might not be such a good idea after all. This definitely should be prototyped as a role on CPAN.
